From bcefffa74be38a8f103f7e82158337819b75adad Mon Sep 17 00:00:00 2001
From: "Guoniu.Zhou" <guoniu.zhou@nxp.com>
Date: Mon, 5 Nov 2018 15:53:14 +0800
Subject: [PATCH 5022/5242] MLK-20230: ov5640: fix read 0x300a fail issue when
 driver probe

commit  4043252cc92aea38d69b76252c6b49f1df1c03c3 from
https://source.codeaurora.org/external/imx/linux-imx.git

Because reset and pwn pin of gpio1 are not configurated for ov5640,
so it leads to read ov5640 register fail.

Signed-off-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../dts/freescale/fsl-imx8qm-mek-mipi-ov5640.dts   |    2 ++
 .../freescale/fsl-imx8qm-mek-mipi-two-ov5640.dts   |    2 ++
 2 files changed, 4 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-mipi-ov5640.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-mipi-ov5640.dts
index 9e4c8e0..df8bf72 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-mipi-ov5640.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-mipi-ov5640.dts
@@ -19,6 +19,8 @@
 		pinctrl_mipi_csi0: mipicsi0grp{
 			fsl,pins = <
 				SC_P_MIPI_CSI0_MCLK_OUT_MIPI_CSI0_ACM_MCLK_OUT	0xC0000041
+				SC_P_MIPI_CSI0_GPIO0_00_LSIO_GPIO1_IO27			0x00000021
+				SC_P_MIPI_CSI0_GPIO0_01_LSIO_GPIO1_IO28			0x00000021
 			>;
 		};
 	};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-mipi-two-ov5640.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-mipi-two-ov5640.dts
index 87e0a57..6f07968 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-mipi-two-ov5640.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-mipi-two-ov5640.dts
@@ -20,6 +20,8 @@
 		pinctrl_mipi_csi1: mipicsi1grp{
 			fsl,pins = <
 				SC_P_MIPI_CSI1_MCLK_OUT_MIPI_CSI1_ACM_MCLK_OUT	0xC0000041
+				SC_P_MIPI_CSI1_GPIO0_00_LSIO_GPIO1_IO30         0x00000021
+				SC_P_MIPI_CSI1_GPIO0_01_LSIO_GPIO1_IO31         0x00000021
 			>;
 		};
 	};
-- 
1.7.9.5

